### DISCUSSION TITLE
Installation of an assembly fails if this contains unsafe blocks

### DISCUSSION THREAD

**Discussion by danielValdezR** [wixtoolset/issues#6731] [Questions] [answered]
Not sure if this is a bug or not.

## Bugs

If this issue is a bug:

* Which version of WiX are you building with?

> 3.11.2.4516

* Which version of Visual Studio are you building with (if any)?

> VS 2022 Professional

* Which version of the WiX Toolset Visual Studio Extension are you building with (if any)?

> 1.0.0.18

* Which version of .NET are you building with?

> .NET Framework 4.8

* If the problem occurs when installing your packages built with WiX, what is the version of Windows the package is running on?

> Windows 10

* Describe the problem and the steps to reproduce it.

> I have a ".wixproj" that installs an assembly which contains the **<AllowUnsafeBlocks>true</AllowUnsafeBlocks>** property. It builds without any issue but when I try to install it, it just doesn't work and shows me the following error in the logs:

> CAQuietExec:  Running a transacted installation.
> CAQuietExec:  
> CAQuietExec:  Beginning the Install phase of the installation.
> CAQuietExec:  See the contents of the log file for the [...]dll assembly's progress.
> CAQuietExec:  The file is located at .
> CAQuietExec:  Installing assembly '[...]dll'.
> CAQuietExec:  Affected parameters are:
> CAQuietExec:     logtoconsole = true
> CAQuietExec:     logfile = 
> CAQuietExec:     assemblypath = [...]dll
> CAQuietExec:  An exception occurred while trying to find the installers in the [...]dll assembly.
> CAQuietExec:  System.Reflection.ReflectionTypeLoadException: Unable to load one or more of the requested types. Retrieve the LoaderExceptions property for more information.
> CAQuietExec:  Aborting installation for [...]dll.
> CAQuietExec:  
> CAQuietExec:  An exception occurred during the Install phase.
> CAQuietExec:  System.InvalidOperationException: Unable to get installer types in the [...]dll assembly.
> CAQuietExec:  The inner exception System.Reflection.ReflectionTypeLoadException was thrown with the following error message: Unable to load one or more of the requested types. Retrieve the LoaderExceptions property for more information..
> CAQuietExec:  
> CAQuietExec:  The Rollback phase of the installation is beginning.

* Describe the behavior you expected and how it differed from the actual behavior.

> This was working well and as expected before I added a new feature to my assembly that required the "unsafeblocks".

**Comment by barnson** [comment, marked as answer]
That's coming from a custom action, not WiX or Windows Installer.

WiX doesn't use issues for support. We moved this issue to the [experimental GitHub discussions area][gh]. You might also consider asking your question on the [wix-users mailing list][ml] or posting on [Stack Overflow][so].

[gh]: https://github.com/wixtoolset/issues/discussions?discussions_q=category%3AQuestions
[ml]: http://wixtoolset.org/documentation/mailinglist/
[so]: https://stackoverflow.com/questions/tagged/wix
